src/charm.py

```diff
@@ -9,28 +9,32 @@
 import logging
 
 from ops.charm import CharmBase
+from ops.framework import StoredState
 from ops.main import main
-# from ops.model import ActiveStatus
+from charms.rolling_ops.v0.rollingops import RollingOpsManager, RollingEvents
 
 logger = logging.getLogger(__name__)
 
 
 class CharmRollingOpsCharm(CharmBase):
     """Charm the service."""
 
+    on = RollingEvents()
```

Comment:
Is this really required here?  It seems like maybe not since the charm itself never binds to any of the rolling events - those are all only used internally by the RollingOpsManager.

Reply:
We do bind stuff to the charm (e.g. charm.on.process_locks). And I don't think that the Framework will pay attention to events that haven't been added to the Charm's offical list 'o events.

I can dig into the internals more, though. I would love to omit this line, especially if I could do so in a way that namespaced all these things.

Reply:
Agree with Robert - there is a way I noted in a comment above :)

Reply:
One thing that I want to do, though I still have to do some more double checking and testing, is namespace all of this. 

So you bind to on[<namespace>].a_library_event

That should get us to a place where we aren't clobbering things.

That said, there is a lot of benefit to having to explicitly modify "on" in your own code. When you do so, you are explicitly verifying that you trust the library to add events. That's nice because it gives charm authors a moment to pause and decide whether or not they trust the library. And it also makes it clear that this library is changing some of the fundamental behavior of the charm ... meaning that you can take advantage of this fact, and bind your own handlers to the library events!